plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/compare/revisionsList/RevisionListPresenter.java

```diff
@@ -0,0 +1,192 @@
+
+        view.setEnableCompareButton(true);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onRevisionDoubleClicked() {
+        compare();
+    }
+
+    /** Get list of revisions. */
+    private void getRevisions() {
+        gitService.log(workspaceId, project, Collections.singletonList(selectedFile), false,
+                       new AsyncRequestCallback<LogResponse>(dtoUnmarshallerFactory.newUnmarshaller(LogResponse.class)) {
+
+                           @Override
+                           protected void onSuccess(LogResponse result) {
+                               view.setRevisions(result.getCommits());
+                               view.showDialog();
+                           }
+
+                           @Override
+                           protected void onFailure(Throwable exception) {
+                               notificationManager.notify(exception.getMessage(), FAIL, false);
+                           }
+                       });
+    }
+
+    private Selection<ResourceBasedNode< ? >> getExplorerSelection() {
+        final Selection<ResourceBasedNode< ? >> selection = (Selection<ResourceBasedNode< ? >>)projectExplorer.getSelection();
+        if (selection == null || selection.isEmpty() || selection.getHeadElement() instanceof HasStorablePath) {
+            return selection;
+        } else {
+            return null;
+        }
+    }
+
+    private void compare() {
+        gitService.diff(workspaceId, project, Collections.singletonList(selectedFile), NAME_STATUS, false, 0, selectedRevision.getId(),
+                        false, new AsyncRequestCallback<String>(new StringUnmarshaller()) {
+                            @Override
+                            protected void onSuccess(String result) {
+                                if (result.isEmpty()) {
+                                    dialogFactory.createMessageDialog(locale.compareMessageIdenticalContentTitle(),
+                                                                      locale.compareMessageIdenticalContentText(), new ConfirmCallback() {
+                                                                          @Override
+                                                                          public void accepted() {
+                                                                              //Do nothing
+                                                                          }
+                                                                      }).show();
+                                } else {
+                                    comparePresenter.show(result.substring(2), result.substring(0, 1), selectedRevision.getId());
+                                }
+                            }
+
+                            @Override
+                            protected void onFailure(Throwable exception) {
+                                notificationManager.notify(exception.getMessage(), FAIL, false);
```

Comment:
Use `org.eclipse.che.ide.ext.git.client.GitLocalizationConstant#diffFailed` instead of `exception.getMessage()`